app/workers/travis_build_killer.rb

```diff
@@ -47,6 +45,12 @@ def kill_builds
           b.cancel
         end
       end
+
+      long_running_jobs.each do |b|
+        next unless b.running?
```

Comment:
Oh, if you cancel it, in between?...that's really confusing...perhaps instead, do the `long_running_jobs =` line after you do the outdated builds?